src/features/blocks/BlocksCategory.tsx

```diff
@@ -0,0 +1,36 @@
+import './BlocksCategory.css'
+import {ToolboxCategory} from "../../toolboxconfig"
+import Block from "./Block"
+
+function BlocksCategory(props: ToolboxCategory) {
+  const categoryStyle = {
+    borderLeft: `10px solid ${props.highlight}`
+  }
+
+  function toggleCategory(e: any) {
+    const categoryBlocksDiv = e.target.nextElementSibling
+    let categoryBlocksDivInnerHeight = 0
+
+    categoryBlocksDiv.childNodes.forEach((block: any) => {
+      categoryBlocksDivInnerHeight += block.clientHeight
+    })
+
+    categoryBlocksDiv.classList.toggle('unrolled')
+    categoryBlocksDiv.style = categoryBlocksDiv.classList.contains('unrolled') ? 'max-height: ' + categoryBlocksDivInnerHeight + 'px' : 'max-height: 0'
```

Comment:
This is alright for quick experimentation, but it's not React-friendly to modify classes and styles of DOM elements directly from the code. The "toggled" state for a category should come from the Redux store. A Redux action should be dispatched when the user clicks a category to toggle it. The Redux action should then use a React reducer to update the state.

Reply:
Thank you very much for your comments and feedback, helped me out a lot.
I've refactored this component's code and will be re-submitting the PR on Monday for a second review.
Thanks again :)